xdsl/parser.py

```diff
@@ -2259,13 +2259,18 @@ def parse_optional_builtin_int_or_float_attr(
         if isinstance(type, IntegerType | IndexType):
             if isinstance(value, float):
                 self.raise_error("Floating point value is not valid for integer type.")
-            return IntegerAttr(value, type)
+            # Pyright workaround, IntegerAttr[IntegerType | IndexType] is not a subtype of
+            # IntegerAttr[IntegerType] | IntegerAttr[IndexType]
+            if isinstance(type, IntegerType):
+                return IntegerAttr(value, type)
+            else:
+                return IntegerAttr(value, type)
```

Comment:
I want to save this in the list of pyright fail highlights

Reply:
It does its best, poor thing